# Отчет

## Импорт библиотек и открытие данных

In [1]:
import numpy as np
from prepare_report import knapsack_report, salesman_report
from prepare_report import res_table_salesman, res_table_knapsack
from data.get_branches import get_salesman, get_knapsack

In [ ]:
!unzip /content/genetic

In [2]:
capacity, profits, weights, optimal = get_knapsack('data/benchmarks/')
graph = get_salesman('data/output_graphs.npy')

## Результаты knapsack 0-1

In [ ]:
total_time = []
for i in range(10):
    time_knapsack, weight_knapsack,profit_knapsack, id_knapsack = knapsack_report(capacity, weights, profits)
    total_time.append(time_knapsack)

In [39]:
total_time = np.array(total_time)
total_time = np.mean(total_time, axis=0)

In [40]:
res_table_knapsack(sorted(capacity.keys()), total_time, 
                   weight_knapsack, profit_knapsack, 
                   id_knapsack)

p01              p02                 p03  \
Weight                             165               26                 190   
Profit                             309               51                 150   
Time                          0.079288         0.061499            0.061475   
ID      [1, 1, 1, 1, 0, 1, 0, 0, 0, 0]  [0, 1, 1, 1, 0]  [1, 1, 0, 0, 1, 0]   

                          p04                       p05  \
Weight                     50                       104   
Profit                    107                       900   
Time                 0.067658                  0.064515   
ID      [1, 0, 0, 1, 0, 0, 0]  [1, 0, 1, 1, 1, 0, 1, 1]   

                          p06                                            p07  
Weight                    169                                            749  
Profit                   1735                                           1458  
Time                 0.067684                                       0.080929  
ID      [0, 1, 0, 1, 0, 0, 1]  [1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1]

**Предыдущая таблица (лаб2)**

| Рюкзаки\Ответ 	| Вес 	| Стоимость 	| Время 	| ID 	|
|---	|---	|---	|---	|---	|
| p01 	| 165   	| 309   	| 0.001335 	| [1, 1, 1, 1, 0, 1, 0, 0, 0, 0] 	|
| p02 	| 26    	| 51    	| 0.000094 	| [0, 1, 1, 1, 0] 	|
| p03 	| 190   	| 150    	| 0.001033  	| [1, 1, 0, 0, 1, 0] 	|
| p04 	| 50    	| 107    	| 0.000238  	| [1, 0, 0, 1, 0, 0] 	|
| p05 	| 104   	| 900   	| 0.000658  	| [1, 0, 1, 1, 1, 0, 1, 1] 	|
| p06 	| 169   	| 1735   	| 0.000863 	| [0, 1, 0, 1, 0, 0, 1] 	|
| p07 	| 749   	| 1458   	| 0.010723 	| [1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1] 	|

## Результаты коммивояжер

In [24]:
(150_000, 10, 1_000, 0.5)

(250000, 5, 900)

In [3]:
time_salesman, weight_salesman, way_salesman, answer_salesman = salesman_report(graph)

100%|██████████| 10/10 [38:57<00:00, 233.74s/it]


In [4]:
res_table_salesman(graph.keys(), time_salesman, 
          weight_salesman, answer_salesman, way_salesman)

gr17  \
Weight                                                  2491.0   
True answer                                             2085.0   
Time                                               2208.527761   
Way          [3, 7, 13, 5, 4, 9, 1, 10, 2, 14, 16, 6, 0, 15...   

                                                          a280  \
Weight                                             2818.621637   
True answer                                             2579.0   
Time                                               2605.435931   
Way          [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,...   

                                                         ch150  \
Weight                                            45292.584585   
True answer                                             6528.0   
Time                                               2309.991053   
Way          [17, 92, 99, 135, 22, 109, 27, 94, 42, 147, 13...   

                                                        bays29  \
Weight                                                  4020.0   
True answer                                             2020.0   
Time                                               2177.366926   
Way          [15, 18, 10, 21, 13, 16, 17, 22, 20, 27, 26, 6...   

                                                         fl417  \
Weight                                            55387.192585   
True answer                                            11861.0   
Time                                                3235.97283   
Way          [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,...   

                                                         att48  
Weight                                                 34141.0  
True answer                                            10628.0  
Time                                               2347.727555  
Way          [3, 47, 12, 31, 10, 16, 2, 46, 14, 11, 21, 38,...

In [5]:
for i in way_salesman:
    print(i)

[3, 7, 13, 5, 4, 9, 1, 10, 2, 14, 16, 6, 0, 15, 11, 8, 12]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 